src/hotspot/os/linux/cgroupV1Subsystem_linux.cpp

```diff
@@ -184,28 +205,28 @@ jlong memory_swap_usage_impl(CgroupController* ctrl) {
   return (jlong)memory_swap_usage;
 }
 
-jlong CgroupV1Subsystem::memory_and_swap_usage_in_bytes() {
-  jlong memory_sw_limit = memory_and_swap_limit_in_bytes();
-  jlong memory_limit = CgroupSubsystem::memory_limit_in_bytes();
+jlong CgroupV1MemoryController::memory_and_swap_usage_in_bytes(julong phys_mem, julong host_swap) {
+  jlong memory_sw_limit = memory_and_swap_limit_in_bytes(phys_mem, host_swap);
+  jlong memory_limit = read_memory_limit_in_bytes(phys_mem);
   if (memory_sw_limit > 0 && memory_limit > 0) {
     jlong delta_swap = memory_sw_limit - memory_limit;
     if (delta_swap > 0) {
-      return memory_swap_usage_impl(_memory->controller());
+      return memory_swap_usage_impl(static_cast<CgroupV1Controller*>(this));
```

Comment:
`memory_swap_usage_impl` could be a method. Likewise for `mem_soft_limit_val`, `mem_swp_limit_val`, `mem_swp_current_val` and `mem_limit_val`.

Reply:
While true those should intentionally be local to the compilation unit. Adding them as methods clutters the code needlessly, IMO.